Frameworks/CoreText/DWriteWrapper_CTFont.mm

```diff
@@ -208,6 +208,173 @@ HRESULT _DWriteCreateFontFaceWithFontDescriptor(CTFontDescriptorRef fontDescript
     return E_INVALIDARG;
 }
 
+// Helper methods for removing fonts that do not match given attributes for CTFontDescriptorCreateMatchingFontDescriptors
+#pragma region Helper Filter Methods
+
+void __FilterFontsByTraits(std::vector<ComPtr<IDWriteFont>>& fonts, CFDictionaryRef traits, bool forceMatch) {
+    _DWriteFontProperties properties = __DWriteFontPropertiesFromTraits(traits);
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    for (auto& font : fonts) {
+        if (font->GetWeight() == properties.weight && font->GetStretch() == properties.stretch && font->GetStyle() == properties.style) {
+            matchingFonts.emplace_back(font);
+        }
+    }
+
+    if (!matchingFonts.empty() || forceMatch) {
+        fonts = std::move(matchingFonts);
+    }
+}
+
+HRESULT __FilterFontsByStyleName(std::vector<ComPtr<IDWriteFont>>& fonts, CFStringRef styleName, bool forceMatch) {
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    for (auto& font : fonts) {
+        ComPtr<IDWriteLocalizedStrings> dwriteFontName;
+        RETURN_IF_FAILED(font->GetFaceNames(&dwriteFontName));
+        CFStringRef name = _CFStringFromLocalizedString(dwriteFontName.Get());
+        if (CFStringCompare(name, styleName, kCFCompareCaseInsensitive) == kCFCompareEqualTo) {
```

Comment:
I think we'd be safe with attribute keys, but I'm not so sure about font names/families, as we might run into problems with "Arial" and "ArialJK" or some other user-defined font nonsense.  Maybe style and display names would be safe? I'm not sure, what do you think?